display_list/display_list.h

```diff
@@ -159,7 +159,12 @@ namespace flutter {
   V(DrawShadowTransparentOccluder)
 
 #define DL_OP_TO_ENUM_VALUE(name) k##name,
-enum class DisplayListOpType { FOR_EACH_DISPLAY_LIST_OP(DL_OP_TO_ENUM_VALUE) };
+enum class DisplayListOpType {
+  FOR_EACH_DISPLAY_LIST_OP(DL_OP_TO_ENUM_VALUE)
+#ifdef IMPELLER_ENABLE_3D
+      DL_OP_TO_ENUM_VALUE(SetSceneColorSource)
+#endif  // IMPELLER_ENABLE_3D
```

Comment:
It seems like a lot of these ifdef blocks could be eliminated in favor of a single ifdef that controls whether or not the FOR_EACH macro contains `V(SetSceneColorSource)` If that `V()` entry doesn't exist then the code isn't generated for that op (and vice versa).